lib/services/key_value_storage/secure_storage_provider.dart

```diff
@@ -0,0 +1,130 @@
+import 'dart:developer';
+import 'dart:io';
+
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:mobileprism/services/key_value_storage/storage_exceptions.dart';
+import 'package:mobileprism/services/key_value_storage/storage_provider.dart';
+
+class SecureStorageProvider implements StorageProvider {
+  final FlutterSecureStorage _storage = const FlutterSecureStorage();
+
+  SecureStorageProvider();
+
+  @override
+  Future<void> storeData(String key, String value) async {
+    log("storeData");
+    log("key: $key");
+    log("value: $value \n");
+    if (!await _storage.containsKey(
+      key: key,
+      aOptions: _getAndroidOptions(),
+      iOptions: _getIOSOptions(),
+    )) {
+      await _storage.write(
+        key: key,
+        value: value,
+        aOptions: _getAndroidOptions(),
+        iOptions: _getIOSOptions(),
+      );
+    } else {
+      throw KeyAlreadyExistsInStorage();
+    }
+  }
```

Comment:
When I try to login by just using a URL the app crashes with an KeyAlreadyExistsInStorage-Exception

Reply:
Can not login at all.